internal/scheduler/context/context.go

```diff
@@ -241,6 +372,56 @@ func (qctx *QueueSchedulingContext) ClearJobSpecs() {
 	}
 }
 
+type GangSchedulingContext struct {
+	Created               time.Time
+	Queue                 string
+	PriorityClassName     string
+	JobSchedulingContexts []*JobSchedulingContext
+	TotalResourceRequests schedulerobjects.ResourceList
+}
+
+func NewGangSchedulingContext(jctxs []*JobSchedulingContext) *GangSchedulingContext {
+	queue := ""
+	for _, jctx := range jctxs {
```

Comment:
What does it mean if some of the job contexts don't have a queue?!   Surely we can just pick the first one.

Reply:
Agreed. I've changed it,